concent_api/core/validation.py

```diff
@@ -410,3 +413,20 @@ def validate_database_report_computed_task(
                 'Nested ReportComputedTask message must be the same as ReportComputedTask stored separately'
             )
         })
+
+
+def substitute_new_report_computed_task_if_needed(report_computed_task: ReportComputedTask, data: bytes) -> Optional[ReportComputedTask]:
+    """
+    If previous sent by requestor report_computed_task does not match report_computed_task sent by provider it
```

Comment:
Apart from mixing Requestor with Provider ;), it needs rephrasing. I would suggest st like this
```python
"""
If stored ReportComputedTask (previously sent by Provider in ForceReportComputedTask message) is different than ReportComputedTask sent by Requestor (in AckReportComputedTask), it means that Provider had sent different  ReportComputedTask to Requestor and the message stored in the DB must be replaced.
"""
```

@rwrzesien. @kbeker, @Jakub89 what do you think guys?